test/parallel/test-timers-clearImmediate.js

```diff
@@ -2,17 +2,39 @@
 require('../common');
 const assert = require('assert');
 
-const N = 3;
-let count = 0;
-function next() {
-  const immediate = setImmediate(function() {
-    clearImmediate(immediate);
-    ++count;
+// run clearImmediate in the callback.
+{
+  const N = 3;
+  let count = 0;
+  function next() {
+    const immediate = setImmediate(function() {
+      clearImmediate(immediate);
+      ++count;
+    });
+  }
+  for (let i = 0; i < N; ++i)
+    next();
+
+  process.on('exit', () => {
+    assert.strictEqual(count, N, `Expected ${N} immediate callback executions`);
   });
 }
-for (let i = 0; i < N; ++i)
-  next();
 
-process.on('exit', () => {
-  assert.strictEqual(count, N, `Expected ${N} immediate callback executions`);
-});
+// run clearImmediate before the callback.
+{
+  const N = 3;
+  let count = 0;
+  function next() {
+    const immediate = setImmediate(function() {
```

Comment:
Now that we have a `common.mustNotCall()`, I think we can use that here and drop `count`.